scripts/machine-launch-script.sh

```diff
@@ -1,83 +1,260 @@
+    CONDA_EXE="${CONDA_INSTALL_PREFIX}/bin/conda"
+
+    if [[ -x "$CONDA_EXE" ]]; then
+	echo "::INFO:: '$CONDA_EXE' already exists, skipping conda install"
+    else
+	wget -O install_conda.sh "$CONDA_INSTALLER"  || curl -fsSLo install_conda.sh "$CONDA_INSTALLER"
+	echo "::INFO:: installing conda to '$CONDA_INSTALL_PREFIX'"
+	# -b for non-interactive install
+	$SUDO bash ./install_conda.sh -b -p "$CONDA_INSTALL_PREFIX"
+    fi
+
+    # see https://conda-forge.org/docs/user/tipsandtricks.html#multiple-channels
+    # for more information on strict channel_priority
+    $SUDO "$CONDA_EXE" config --system --set channel_priority strict
+    # By default, don't mess with people's PS1, I personally find it annoying
+    $SUDO "$CONDA_EXE" config --system --set changeps1 false
+
+    # https://conda-forge.org/feedstock-outputs/ 
+    #	filterable list of all conda-forge packages
+    # https://conda-forge.org/#contribute 
+    #	instructions on adding a recipe
+    # https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications
+    #   documentation on package_spec syntax
+    CONDA_PACKAGE_SPECS=()
+
+
+    # https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#using-the-compiler-packages
+    #    for notes on using the conda compilers
+    CONDA_PACKAGE_SPECS=( gcc_linux-64 gxx_linux-64 )
+
+    # if we want to install prebuilt riscv toolchain, the bdutrosv channel has riscv-gnu-toolchain
+
+    # poky deps
+    CONDA_PACKAGE_SPECS+=( python=3.8 patch texinfo subversion chrpath git wget )
+    # qemu deps
+    CONDA_PACKAGE_SPECS+=( gtk3 glib pkg-config bison flex )
+    # qemu also requires being built against kernel-headers >= 2.6.38 because of it's use of
+    # MADV_NOHUGEPAGE in a call to madvise.  See:
+    # https://man7.org/linux/man-pages/man2/madvise.2.html
+    # https://conda-forge.org/docs/maintainer/knowledge_base.html#using-centos-7
+    # obvi this would need to be made linux-specific if we supported other MacOS or Windows
+    CONDA_PACKAGE_SPECS+=( "kernel-headers_linux-64>=2.6.38" )
+    # firemarshal deps
+    CONDA_PACKAGE_SPECS+=( rsync )
+    # cross-compile glibc 2.28+ deps
+    CONDA_PACKAGE_SPECS+=( make )
+    # other misc deps
+    # conda-forge splits out the shared library into libzlib
+    # not sure if we need both, but I'm including them for now
+    CONDA_PACKAGE_SPECS+=(
+        sbt \
+        ca-certificates \
+        mosh \
+        gmp \
+        mpfr \
+        mpc \
+        zlib \
+	libzlib \
+        vim \
+        git  \
+        openjdk \
+        gengetopt \
+        libffi \
+        expat \
+        libusb1 \
+        ncurses \
+        cmake \
+        graphviz \
+        expect
+        dtc \
+    )
+
+
+    # Why do we need these?
+    #   "perl(ExtUtils::MakeMaker)" \
+    # diffstat texi2html 
+
+    # verilator was added to conda-forge by timsnyder
+    CONDA_PACKAGE_SPECS+=( verilator==4.034 )
```

Comment:
Are you wanting to support building only Verilator from source in this script or any of the packages?  Just trying to understand the scope of what you're suggesting.

Reply:
In theory, now that I have the verilator builds working again, any new versions that come out should be available within a day or two of them being released by the verilator guy and it would be as simple as just updating the version here.  I only say "in theory" because I was lazy between creating 4.034 and a few weeks ago and didn't bump for each version that came out.

Reply:
Hmm. I was thinking if any package (Verilator or otherwise) needs to be built from source we can do that with a special hook. But the cleaner option is to just enforce that people add things to `conda-forge` before it can be added to this file. 

To make sure I understand, you added some automation that auto-builds new Verilator versions and adds it to `conda-forge`? Thats very nice.

Reply:
conda-forge has this automation for all packages and I'm committing to paying attention to new PR's that are opened in the verilator-feedstock repo for version bumps if we start using this.  The repo is https://github.com/conda-forge/verilator-feedstock if you are interested.  I guess there was a new PR opened two days ago that I hadn't noticed and I'm going to go now and look to make sure it seems okay and merge it so that the new version becomes available on anaconda.org. https://github.com/conda-forge/verilator-feedstock/pull/15

I think the cleanest thing for any conda package would be to contribute to bumping the feedstock recipe so that the new version gets built.  The easiest way to find the feedstock repos for a given package is on the conda-forge package listing at https://conda-forge.org/feedstock-outputs/. Each package there has a link to it's page on anaconda.org (where you can see the existing versions) and a link to it's feedstock repo (where you can see any new versions that are being worked on or submit a PR to build a different version.

I say it is "cleanest" because you already have a working "recipe" for doing the build and all of the dependencies captured.  Using the conda-forge CI to do the build will build it in their isolated docker images and you can have some confidence 

If you want to quickly do a build just to try things out, you can clone the feedstock repo, modify the `recipe/meta.yaml` to point to the new version and use the `./build-locally.py` to run the build in a local docker container and generate packages that you can install into your conda env without necessarily uploading them to anaconda.org.

Reply:
Oh, and the reason, it matters that I pay attention to verilator, is because I'm currently the only maintainer of the conda recipe...

Reply:
Perhaps I should add a page to the documentation that explains how to do all of what I said above with pointers into the conda-forge docs?

Reply:
@davidbiancolin and @abejgonzalez, I pushed the new page in the docs.